keps/sig-cloud-provider/20191004-out-of-tree-credential-providers.md

```diff
@@ -131,15 +135,21 @@ type RegistryCredentialProvider struct {
     // contains a port, then the port must match as well.
     ImageMatchers []string `json:"imageMatchers"`
 
-    // Exec specifies a custom exec-based plugin.  This type is defined in
-    // https://github.com/kubernetes/client-go/blob/62f256057db7571c5ed1aba47eea291f72dd557a/tools/clientcmd/api/types.go#L184
-    Exec clientcmd.ExecConfig
+    // ExtraArgs specifies arguments to be passed to the plugin via environment variables.
+    // +optional
+    ExtraArgs []PluginArg `json:"extraArgs"`
+}
+
+// PluginArg is used for passing extra arguments to the exec credential plugin binary.
+type PluginArg struct {
+    Name  string `json:"name"`
+    Value string `json:"value"`
 }
 ```
 
 The RegistryCredentialConfig will be encoded in YAML and located in a file on disk.  The exact path of the credential provider configuration file will be passed to kubelet via a new configuration option `RegistryCredentialConfigPath`.
 
-We will create new types `RegistryCredentialPluginRequest` and `RegistryCredentialPluginResponse` which will define the interface between the plugin and the kubelet runtime.  After the kubelet matches the image property string to a RegistryCredentialProvider, the kubelet will exec the plugin binary, and pass the JSON encoded request to the plugin via stdin.  This includes the image that is to be pulled.  The plugin will report back the response, which includes the credentials that kubelet needs to pull the image.
+We will create new types `RegistryCredentialPluginRequest` and `RegistryCredentialPluginResponse` which will define the interface between the plugin and the kubelet runtime.  After the kubelet matches the image property string to a RegistryCredentialProvider, the kubelet will exec the plugin binary, providing the argument get-credentials, and pass the JSON encoded request to the plugin via stdin, which includes the image that is to be pulled.  The plugin will report back the response, which includes the credentials that kubelet needs to pull the image.
```

Comment:
> the kubelet will exec the plugin binary, providing the argument get-credentials

Wondering if there should be a separate section specifying the standard exec arguments any given plugin must  implement. Today it would just be:
```
<plugin> get-credentials arg1=value1,arg2=value2,.... 
```